File: samples/VB .NET Samples/TheMinimumSystem/TheMinimumSystem/SourceListWrapper.vb
Imports System.Collections.Generic
Imports System.Text

Public Class SourceListWrapper
	Private m_listSourceNames As List(Of String) = Nothing
	Public Sub New(listSourceNames As List(Of String))
		m_listSourceNames = listSourceNames
	End Sub

	Private m_SelectedIndex As Integer = 0
	Public Function SelectSource() As Integer
		Dim temp As New SourceListForm(m_listSourceNames)
		temp.ShowDialog()
		m_SelectedIndex = temp.GetSelectedIndex()
		Return m_SelectedIndex
	End Function

End Class

File: samples/C# Samples/SeparateDocumentByBarcode/SeparateDocumentByBarcode/SourceListWrapper.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SeparateDocumentByBarcode
{
    public class SourceListWrapper
    {
        private List<string> m_listSourceNames = null;
        public SourceListWrapper(List<string> listSourceNames)
        {
            m_listSourceNames = listSourceNames;
        }

        private int m_SelectedIndex = 0;
        public int SelectSource()
        {
            SourceListForm temp = new SourceListForm(m_listSourceNames);
            temp.ShowDialog();
            m_SelectedIndex = temp.GetSelectedIndex();
            return m_SelectedIndex;
        }

    }
}
